kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -90,33 +107,39 @@ public void close() {
 
     public void close(int code, String reason) {
         try {
-         closeExecutor();
-         closeWebSocket(code, reason);
+          cleanupOnClose.set(true);
+          closeWebSocket(code, reason);
         } finally {
           if (listener != null) {
             listener.onClose(code, reason);
           }
         }
     }
 
-    private void closeExecutor() {
-      if (!executorClosed.compareAndSet(false, true)) {
+  /**
+   * Performs the cleanup tasks:
+   * 1. closes the InputStream pumper
+   * 2. closes all internally managed closeables (piped streams).
+   *
+   * The order of these tasks can't change or its likely that the pumper will through errors,
+   * if the stream it uses closes before the pumper it self.
+   */
+  private void cleanUp() {
```

Comment:
As far as I can tell, cleanUp() is called only if `ExecWatch.close()` is called. I feel like at least some clean up should be done if the WebSocket we're listening on is closed (e.g. close the write end of the output pipes).